### DISCUSSION TITLE
Binding child components using an Array or Collection

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [livewire/livewire#1038] [4. General]
When nesting components, can a parent component bind `array` or `Collection` variables to render child components? Like showing a Manager screen with a list of Employees under a manager and the Employee component that takes an array object to bind data to render the component.

**Comment by mblarsen** [comment]
Livewire does not allow model binding (not sure what the reason is. So you have in your mount method extract props on the model to public props on the component. E.g. `$this->name = $employee->name`. However, it does support plain old arrays. So you can do this:

```php
public $employee;

public function mount(Employee $employee)
{
    $this->employee = $employ->toArray();
}
```

And then in your component using dot-notation:

```html
<input wire:model="employee.name"/>
```

Or if you don't want to use nested components (you can still use Laravel components or partials):

```html
@foreach($employees as $employee)
<input wire:model="employees.{{$index}}.name"/>
@endforeach
```
(remember that `employees` should be an array)

---
Maybe I misunderstood your problem. In that case please elaborate and I'll see if I can help.

**Reply by abishekrsrikaanth** [reply]
Thanks, I think I can work with the array object. It would definitely be good to be able to pass around serializable objects or models to components.

**Reply by mblarsen** [reply]
Agreed. If I figure out why that isn't possible I'll let you know :)
